internal/xds/bootstrap/bootstrap.go

```diff
@@ -138,6 +154,25 @@ func GetRenderedBootstrapConfig(proxyMetrics *egcfgv1a1.ProxyMetrics) (string, e
 				Port:    sink.OpenTelemetry.Port,
 			})
 		}
+
+		if proxyMetrics.Matches != nil {
+			enableStatConfig = true
+			// Set default envoy proxy Stats
+			StatsMatcher.Prefixs = append(StatsMatcher.Prefixs, strings.Split(DefaultEnvoyStatsMatcherPrefixes, ",")...)
```

Comment:
why are we also adding `DefaultEnvoyStatsMatcherPrefixes` here ?
imo we should ONLY we adding `proxyMetrics.Matches`

Reply:
comment same as above

Reply:
How about enabling only the warming cluster stats for this case ? And adding a comment why in docs

Reply:
Of course we can only expose warming cluster stats，but I think these stats(`DefaultEnvoyStatsMatcherPrefixes = "cluster_manager,listener_manager,server,cluster.xds_cluster"`) are very useful.

- [cluster_manager(https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats#general):  `cluster_manager` can view the cluster size of the current proxy
- [listener_manager:](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/stats) `listener_manager` can view the cluster size of the current proxy
- cluster.xds_cluster: `cluster.xds_cluster` can view the interaction with the control plane gateway
- [server:](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/statistics)  Server related statistics are rooted at server

Reply:
the API you are implementing is an opt in API
why do we need to enable all other stats by default ? if the user wanted all stats they wouldn't use this API, since not setting a match enables all stats
the only reason to enable the warming cluster stats when a match has been defined is to make sure the critical functionality in system doesnt break

Reply:
Maybe other default stats can be added later if more users need it later.

Reply:
let's keep it empty for now.
if it's really needed, it can be done with a seperated PR.

Reply:
lets go with empty as @zirain suggested with a big disclaimer in the API that some stats are needed for basic functioning
relates to
https://github.com/envoyproxy/envoy/issues/9856
https://github.com/envoyproxy/envoy/pull/9883
https://github.com/envoyproxy/envoy/issues/14610

we can also add a match exclusion API field to opt out instead in the near future

Reply:
thoughts on updating the API to
```
metrics:
  matches:
  - include:
       type: "Prefix"
       value: ...
  - exclude:
       type: "Suffix"
       value: ....
 ```
and we can add disclaimer saying that `reducing stats helps reduce memory consumption but
eliminating some stats may after critical functionality, here are the stats that we strongly recommend not disabling .......`

Reply:
I agree, adding a disclaimer is appropriate.
The idea of ​​updating the API is ok to me.

Reply:
Let's continue with this PR for now, and leave API as

Reply:
done